diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1a0e09d4..af9687b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,6 +102,8 @@ endif()
 # Use our cmake files
 list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
+list(INSERT CMAKE_FIND_ROOT_PATH 0 "${ADDITIONAL_FIND_ROOT_PATH}")
+
 # Require Windows 7
 if(WIN32)
   set(WINVER 0x0601)
@@ -114,8 +116,8 @@ if(USE_ICU)
     include(winicu)
     EnableWinICU()
   else()
-    find_package(ICU REQUIRED COMPONENTS uc i18n)
-    link_libraries(ICU::uc ICU::i18n)
+    find_package(ICU REQUIRED COMPONENTS uc i18n data)
+    link_libraries(ICU::uc ICU::i18n ICU::data)
   endif()
 endif()
 
@@ -256,18 +258,6 @@ if(WIN32)
   link_libraries("vfw32" "comctl32" "winmm")
 endif()
 
-if(ENABLE_SDL OR ENABLE_WIN OR ENABLE_TOOLS)
-  if(ENABLE_GLES)
-    find_package(OpenGLES COMPONENTS V2 REQUIRED)
-  else()
-    find_package(OpenGL REQUIRED)
-  endif()
-endif()
-
-find_package(Libepoxy REQUIRED)
-link_libraries(libepoxy::libepoxy)
-include_directories(${LIBEPOXY_INCLUDE_DIR})
-
 find_package(Eigen3 3.3 REQUIRED NO_MODULE) # -DEigen3_DIR=...
 message(STATUS "Found Eigen3 ${EIGEN3_VERSION_STRING}")
 link_libraries(Eigen3::Eigen)
diff --git a/src/celestia/CMakeLists.txt b/src/celestia/CMakeLists.txt
index bfef78b1..2e38421f 100644
--- a/src/celestia/CMakeLists.txt
+++ b/src/celestia/CMakeLists.txt
@@ -95,7 +95,7 @@ list(GET PROJECT_VERSION_LIST 0 PROJECT_VERSION_MAJOR)
 list(GET PROJECT_VERSION_LIST 1 PROJECT_VERSION_MINOR)
 set(CELSO "celestia-${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}")
 
-add_library(celestia SHARED ${CELESTIA_SOURCES}
+add_library(celestia STATIC ${CELESTIA_SOURCES}
                             ${SCRIPT_LIBS}
                             ${CELESTIA_CORE_LIBS}
 )
diff --git a/src/celestia/sdl/CMakeLists.txt b/src/celestia/sdl/CMakeLists.txt
index e4015ba3..b91fce60 100644
--- a/src/celestia/sdl/CMakeLists.txt
+++ b/src/celestia/sdl/CMakeLists.txt
@@ -30,9 +30,6 @@ set(IMGUI_SOURCES
 add_library(imgui STATIC ${IMGUI_SOURCES})
 if(ENABLE_GLES)
   target_compile_definitions(imgui PUBLIC IMGUI_IMPL_OPENGL_ES2)
-  target_link_libraries(imgui PRIVATE OpenGLES::OpenGLESv2)
-else()
-  target_link_libraries(imgui PRIVATE OpenGL::GL)
 endif()
 
 target_include_directories(imgui PUBLIC "${IMGUI_DIR}" "${IMGUI_DIR}/backends" "${IMGUI_DIR}/misc/cpp")
@@ -82,6 +79,8 @@ else()
   target_link_libraries(imgui PUBLIC ${SDL2_LIBRARIES})
 endif()
 
+set_target_properties(celestia-sdl PROPERTIES LINK_FLAGS "--preload-file ${CMAKE_SOURCE_DIR}/CelestiaResources@/ --bind -sUSE_WEBGL2=1 -sGL_ENABLE_GET_PROC_ADDRESS -sTOTAL_MEMORY=419430400 --js-library ${CMAKE_SOURCE_DIR}/library.js")
+
 install(
   TARGETS celestia-sdl
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
