diff --git a/library.js b/library.js
new file mode 100644
index 00000000..37651210
--- /dev/null
+++ b/library.js
@@ -0,0 +1,9 @@
+addToLibrary({
+    celestiaLoadSuccess: function() {
+        loadSuccessCallback();
+    },
+    celestiaLoadFile: function(file, lenfile) {
+        const fileName = UTF8ToString(file, lenfile);
+        loadFileCallback(fileName);
+    },
+});
\ No newline at end of file
diff --git a/src/celestia/sdl/appwindow.cpp b/src/celestia/sdl/appwindow.cpp
index 3752af0f..72b2c12e 100644
--- a/src/celestia/sdl/appwindow.cpp
+++ b/src/celestia/sdl/appwindow.cpp
@@ -30,6 +30,10 @@
 #include "gui.h"
 #include "settings.h"

+extern "C" {
+    extern void celestiaLoadSuccess();
+}
+
 namespace celestia::sdl
 {

@@ -191,6 +195,9 @@ AppWindow::run(const Settings& settings)

     m_appCore->start();
     setTimezone(*m_appCore);
+
+    celestiaLoadSuccess();
+
     SDL_GL_GetDrawableSize(m_window.get(), &m_width, &m_height);
     m_appCore->resize(m_width, m_height);

diff --git a/src/celestia/sdl/sdlmain.cpp b/src/celestia/sdl/sdlmain.cpp
index cf15e661..f4e29d38 100644
--- a/src/celestia/sdl/sdlmain.cpp
+++ b/src/celestia/sdl/sdlmain.cpp
@@ -13,6 +13,7 @@
 #include <utility>

 #include <celestia/celestiacore.h>
+#include <celestia/progressnotifier.h>
 #include <celutil/gettext.h>
 #include "alerter.h"
 #include "appwindow.h"
@@ -24,9 +25,25 @@
 #include <windows.h>
 #endif

+extern "C" {
+    extern void celestiaLoadFile(const char *file, size_t lenfile);
+}
+
 namespace
 {

+class CelestiaCoreProgressWatcher: public ProgressNotifier
+{
+public:
+    CelestiaCoreProgressWatcher() : ProgressNotifier() {}
+
+    void update(const std::string& status)
+    {
+        fmt::print("Loading: {}\n", status);
+        celestiaLoadFile(status.c_str(), status.size());
+    }
+};
+
 std::filesystem::path
 getDataDir()
 {
@@ -59,9 +76,9 @@ main(int argc, char **argv)
     CelestiaCore::initLocale();

 #ifdef ENABLE_NLS
-    bindtextdomain("celestia", LOCALEDIR);
+    bindtextdomain("celestia", "locale");
     bind_textdomain_codeset("celestia", "UTF-8");
-    bindtextdomain("celestia-data", LOCALEDIR);
+    bindtextdomain("celestia-data", "locale");
     bind_textdomain_codeset("celestia", "UTF-8");
     textdomain("celestia");
 #endif
@@ -74,19 +91,13 @@ main(int argc, char **argv)
     auto alerter = std::make_unique<celestia::sdl::Alerter>();
     celestiaCore->setAlerter(alerter.get());

-    if (!celestiaCore->initSimulation())
+    CelestiaCoreProgressWatcher progressWatcher;
+    if (!celestiaCore->initSimulation("celestia.cfg", {}, &progressWatcher))
         return EXIT_FAILURE;

     if (!environment->setGLAttributes(celestiaCore->getConfig()->renderDetails.aaSamples))
         return EXIT_FAILURE;

-    std::filesystem::path dataDir = getDataDir();
-
-    std::error_code ec;
-    std::filesystem::current_path(dataDir, ec);
-    if (ec)
-        return EXIT_FAILURE;
-
     Settings settings = Settings::load(environment->getSettingsPath());

     auto window = environment->createAppWindow(settings, std::move(celestiaCore), std::move(alerter));
